packages/react-components/react-utilities/src/hooks/useFirstMount.ts

```diff
@@ -14,10 +15,9 @@ import * as React from 'react';
 export function useFirstMount(): boolean {
   const isFirst = React.useRef(true);
 
-  if (isFirst.current) {
+  useIsomorphicLayoutEffect(() => {
```

Comment:
Can you please clarify why it's needed? React also unmounts `React.useRef()`, so a value from the first render will be dismissed and the current implementation should work properly:

https://codesandbox.io/s/old-frog-dnhi1v?file=/src/App.js

Reply:
Yeah I was just remembering the best practices which stated writing refs during render is unsafe. Good to know that in this case, it doesn't actually matter